linera-service/tests/linera_net_tests.rs

```diff
@@ -2980,3 +3004,77 @@ async fn test_end_to_end_listen_for_new_rounds(config: impl LineraNetConfig) ->
 
     Ok(())
 }
+
+/// Tests token transfers between two chains and measures the latency.
+///
+/// To use this as a benchmark, make sure to run it with `--nocapture`, e.g.:
+///
+/// ```bash
+/// RUST_LOG=info cargo test -p linera-service \
+///     --features storage-service
+///     test_end_to_end_repeated_transfers::storage_service_grpc \
+///     -- --nocapture
+/// ```
+///
+/// It will print the average end-to-end transfer latency, including creating the sending block,
+/// the receiving block, and the resulting `NewBlock` notification.
+#[cfg_attr(feature = "storage-service", test_case(LocalNetConfig::new_test(Database::Service, Network::Grpc) ; "storage_service_grpc"))]
+#[cfg_attr(feature = "scylladb", test_case(LocalNetConfig::new_test(Database::ScyllaDb, Network::Grpc) ; "scylladb_grpc"))]
+#[cfg_attr(feature = "dynamodb", test_case(LocalNetConfig::new_test(Database::DynamoDb, Network::Grpc) ; "aws_grpc"))]
+#[cfg_attr(feature = "kubernetes", test_case(SharedLocalKubernetesNetTestingConfig::new(Network::Grpc, BuildArg::Build) ; "kubernetes_grpc"))]
+#[cfg_attr(feature = "remote-net", test_case(RemoteNetTestingConfig::new(None) ; "remote_net_grpc"))]
+#[test_log::test(tokio::test)]
+async fn test_end_to_end_repeated_transfers(config: impl LineraNetConfig) -> Result<()> {
+    let _guard = INTEGRATION_TEST_GUARD.lock().await;
+    tracing::info!("Starting test {}", test_name!());
+
+    let transfer_count = test_iterations().unwrap_or(100);
+
+    let (mut net, client1) = config.instantiate().await?;
+    let chain1 = client1.load_wallet()?.default_chain().unwrap();
+    let client2 = net.make_client().await;
+    client2.wallet_init(&[], FaucetOption::None).await?;
+    let chain2 = client1.open_and_assign(&client2, Amount::ONE).await?;
+    let node_service2 = client2
+        .run_node_service(None, ProcessInbox::Automatic)
+        .await?;
+    let mut notifications2 = Box::pin(node_service2.notifications(chain2).await.unwrap());
+
+    let start_time = Instant::now();
+    for i in 0..transfer_count {
+        client1
+            .transfer(Amount::from_attos(1), chain1, chain2)
+            .await
+            .unwrap();
+        loop {
+            let sleep = Box::pin(linera_base::time::timer::sleep(Duration::from_secs(5)));
```

Comment:
5 seconds feels excessive?

Reply:
These are only awaited if there is no notification; they don't slow down the test in the success case.

I think it's fine to wait for five seconds before calling it a failure?

Reply:
I was just thinking that 1 or 2 seconds delay might be worth examining. Either way, LGTM!

Reply:
Good point, I'll decrease it to 1 second. If that makes it flaky we'll increase it.

Also, I realized I shouldn't restart the timer each time we receive an irrelevant notification.